File: content/cpp/templates/01_Basics/main.cpp
/**
 * Main Program Template
 * Entry point for C++ application
 */
#include <iostream>
#include <string>
using namespace std;

int main() {
    // Welcome message
    cout << "Welcome to the C++ program!" << endl;
    
    // Read user input
    cout << "Enter your name: ";
    string name;
    getline(cin, name);
    
    // Display greeting
    cout << "Hello, " << name << "!" << endl;
    
    return 0;
}

File: content/java/templates/01_Basics/main.java
/**
 * Main Class Template
 * Entry point for Java application
 */
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        // Create scanner for user input
        Scanner scanner = new Scanner(System.in);
        
        // Welcome message
        System.out.println("Welcome to the Java program!");
        
        // Read user input
        System.out.print("Enter your name: ");
        String name = scanner.nextLine();
        
        // Display greeting
        System.out.println("Hello, " + name + "!");
        
        // Close scanner
        scanner.close();
    }
}
